crypto/dh/dh_check.c

```diff
@@ -102,10 +99,11 @@ int DH_check(const DH *dh, int *ret)
 {
     int ok = 0, r;
     BN_CTX *ctx = NULL;
-    BN_ULONG l;
     BIGNUM *t1 = NULL, *t2 = NULL;
 
-    *ret = 0;
+    if (!DH_check_params(dh, ret))
```

Comment:
please look here.

Reply:
*ret = 0; is removed, and DH_check_params used to initialize *ret.
So DH_check_params returns 1, but *ret is already DH_NOT_SUITABLE_GENERATOR.